api/kyverno/v2alpha1/policy_exception.go

```diff
@@ -0,0 +1,64 @@
+package v2alpha1
+
+import (
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PolicyException declares resources to be excluded from specified policies.
+type CELPolicyException struct {
+	metav1.TypeMeta   `json:",inline,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec declares policy exception behaviors.
+	Spec CELPolicyExceptionSpec `json:"spec"`
+}
+
+func (p *CELPolicyException) GetKind() string {
+	return "CELPolicyException"
+}
+
+// Validate implements programmatic validation
+func (p *CELPolicyException) Validate() (errs field.ErrorList) {
+	errs = append(errs, p.Spec.Validate(field.NewPath("spec"))...)
+	return errs
+}
+
+// PolicyExceptionSpec stores policy exception spec
+type CELPolicyExceptionSpec struct {
+	// PolicyNames identifies the policies to which the exception is applied.
+	PolicyNames []string `json:"policyNames"`
+
+	// MatchConstraints is used to check if a resource applies to the exception.
+	MatchConstraints *admissionregistrationv1.MatchResources `json:"matchConstraints"`
```

Comment:
Do we need match constraints ?

Reply:
It can be optional IMHO.

Reply:
I think we need it to check if the resource matches the exception, especially for autogen-rules. But, we can mark it as optional.

Reply:
You can clearly do the same using match conditions, is there any benefit supporting both ?

Reply:
In case we have autogen rules, and we need to exclude Deployments, it is possible to check that the object is deployment in the match conditions. But, when we generate a VAP, the exceptions' match conditions will be translated to VAP's match condition with a negative operator. In that case, it wouldn't make sense to have a match condition in VAP that checks if the resource is Deployment, as matchConstraints specify Pods only. WDYT?

We can remove it for now, and add it later if needed.

Reply:
In addition, match resources in exceptions will be translated to exclude resources in VAPs. For example, we can have a policy that matches Pods, and an exception that matches the `test-ns` Namespace. This will generate a VAP that excludes the `test-ns` Namespace using the `excludeResourceRules`. Another solution is to write a match condition to check the resource's namespace. I think we should support both options.

Reply:
Let's se what we need at implementation time, just keep in mind that some stuff will deserve some cleanup.